tests/func/utils/test_strict_yaml.py

```diff
@@ -0,0 +1,310 @@
+import os
+
+import pytest
+from ruamel.yaml import __with_libyaml__ as ruamel_clib
+
+from dvc.main import main
+from dvc.ui import ui
+
+DUPLICATE_KEYS = """\
+stages:
+  stage1:
+    cmd: python train.py
+    cmd: python train.py
+"""
+
+DUPLICATE_KEYS_OUTPUT = """\
+'./dvc.yaml' is corrupted.
+
+While constructing a mapping, in line 3, column 5
+  3 │   cmd: python train.py
+
+Found duplicate key "cmd" with value "python train.py" (original value:\
+ "python \ntrain.py"), in line 4, column 5
+  4 │   cmd: python train.py"""
+
+
+MAPPING_VALUES_NOT_ALLOWED = """\
+stages:
+  stage1
+    cmd: python script.py
+"""
+
+MAPPING_VALUES_NOT_ALLOWED_OUTPUT = """\
+'./dvc.yaml' is corrupted.
+
+Mapping values are not allowed {}, in line 3, column 8
+  3 │   cmd: python script.py""".format(
+    "in this context" if ruamel_clib else "here"
+)
+
+
+NO_HYPEN_INDICATOR_IN_BLOCK = """\
+stages:
+  stage1:
+    cmd: python script.py
+    outs:
+      - logs:
+          cache: false
+      metrics:
+"""
+
+NO_HYPEN_INDICATOR_IN_BLOCK_OUTPUT = """\
+'./dvc.yaml' is corrupted.
+
+While parsing a block collection, in line 5, column 7
+  5 │     - logs:
+
+{}, in line 7, column 7
+  7 │     metrics:""".format(
+    "Did not find expected '-' indicator"
+    if ruamel_clib
+    else "Expected <block end>, but found '?'"
+)
+
+
+UNCLOSED_SCALAR = """\
+stages:
+  stage1:
+    cmd: python script.py
+    desc: "this is my stage one
+"""
+
+UNCLOSED_SCALAR_OUTPUT = """\
+'./dvc.yaml' is corrupted.
+
+While scanning a quoted scalar, in line 4, column 11
+  4 │   desc: "this is my stage one
+
+Found unexpected end of stream, in line 5, column 1
+  5"""
+
+
+NOT_A_DICT = "3"
+NOT_A_DICT_OUTPUT = "'./dvc.yaml' validation failed: expected a dictionary.\n"
+
+
+MISSING_CMD = """\
+stages:
+  stage1:
+"""
+
+MISSING_CMD_OUTPUT = """\
+'./dvc.yaml' validation failed.
+
+expected a dictionary, in stages -> stage1, line 2, column 3
+  1 stages:
+  2   stage1:
+  3"""
+
+
+DEPS_AS_DICT = """\
+stages:
+  stage1:
+    cmd: python script.py
+    deps:
+      - src:
+"""
+
+DEPS_AS_DICT_OUTPUT = """\
+'./dvc.yaml' validation failed.
+
+expected str, in stages -> stage1 -> deps -> 0, line 5, column 9
+  4 │   deps:
+  5 │     - src:
+"""
+
+NULL_VALUE_ON_OUTS = """\
+stages:
+  cmd: python script.py
+  outs:
+  - logs:
+      cache: false
+      persist: true
+      remote:
+"""
+
+NULL_VALUE_ON_OUTS_OUTPUT = """\
+'./dvc.yaml' validation failed: 2 errors.
+
+expected a dictionary, in stages -> cmd, line 2, column 3
```

Comment:
Why are this one and the next few missing a stage name?

Reply:
🤦🏼‍♂️